### ISSUE TITLE
Jackson

### ISSUE THREAD

**Issue by mhoffman78** [xtivia/dxp-rest-example#2]
Hello,

I am attempting to get this example to work and I am running to an error where Tomcat will not recognize the fasterxml package

"Unresolved requirement: Import-Package: com.fasterxml.jackson.jaxrs.json; version="[2.9.0,3.0.0)"_ [Sanitized]"

I have the following in my build.gradle and it is listed in my Import-package within the MANIFEST.MF
**build.gradle**
compileInclude	group: "com.fasterxml.jackson.jaxrs", name: "jackson-jaxrs-json-provider", version: "2.9.3"

**MANIFEST.MF**
Import-Package: aQute.bnd.annotation.metatype;version="[1.45,2)",com.fasterxml.jackson.jaxrs.json;version="[2.9,3)"...

I have also tried CompileOnly within the build.gradle and that did not work either

Any thoughts?

**Comment by mfreeman-xtivia** [comment]
Are the Jackson JARs being included inside your generated JAR (the one that ends up in the gradle build/libs directory). It sounds like they're not, and at deploy time its trying to find them inside the OSGi container environment, which it wont have any luck with.

If they're being included correctly they should be inside the OSGI-INF/lib folder of the JAR

**Comment by mhoffman78** [comment]
Yes the following jars are inside my generated jar under root/lib

- jackson-annotations-2.9.0.jar
- jackson-core-2.9.3.jar
- jackson-databind-2.9.3.jar
- jackson-jaxrs-base-2.9.3.jar
- jackson-jaxrs-json-provider-2.9.3.jar
- jackson-module-jaxb-annotations-2.9.3.jar

**Comment by mfreeman-xtivia** [comment]
If they are truly in root/lib its not likely to work. They should be under OSGi-INF/lib.

I will pull down the current version from GitHub tonight and take a look.

**Comment by mhoffman78** [comment]
I do not have a OSGI-INF/lib.  How would I get one?

**Comment by mfreeman-xtivia** [comment]
lines 43-62 of build.gradle cause that to happen

**Comment by mhoffman78** [comment]
Will the jar task in your sample build.gradle override the one in included in Liferay?

**Comment by mfreeman-xtivia** [comment]
Sorry dont understand the question, in particular "the one included in Liferay"

**Comment by mhoffman78** [comment]
There is a "jar" gradle task already defined in the Liferay included build.gradle.  When you create a Service Builder projects the gradle tasks are already defined

**Comment by mfreeman-xtivia** [comment]
This project has zero to do with any other Liferay style projects, including service build projects. Its intended to be used on its own to build REST services that you implement in a custom fashion. If you're thinking in terms of JSON-WS services for service builder projects from the 6.x days this is totally different.

**Comment by mhoffman78** [comment]
I am trying to stand up a JAX-RS web service to call an underlying service that is built using the Service Builder in DXP.

**Comment by angelcazares** [comment]
Have you tried putting into your bnd.bnd file your dependencies like this:
Include-Resource: @jackson-annotations-2.8.8.jar, @jackson-core-2.8.8.jar, @jackson-databind-2.8.8.jar

Or have you tried putting the following in your gradle file:

compileInclude group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.8.8'
compileInclude group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.8.8'
compileInclude group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.8.8'
